### ISSUE TITLE
crash when adding enum type

### ISSUE THREAD

**Issue by lucasduffey** [Vector35/binaryninja-api#785]
# repro
Ubuntu 16 LTS on dev branch

using the UI I added type name "compression_type" with the following type data:
```
enum {
    COMP_STORED    = 0,
    COMP_SHRUNK    = 1,
    COMP_REDUCED1  = 2,
    COMP_REDUCED2  = 3,
    COMP_REDUCED3  = 4,
    COMP_REDUCED4  = 5,
    COMP_IMPLODED  = 6,
    COMP_TOKEN     = 7,
    COMP_DEFLATE   = 8,
    COMP_DEFLATE64 = 9
}
```

# backtrace
```
(gdb) 
#0  0x00007ffff63dd7c3 in ?? () from /home/user/binaryninja/libbinaryninjacore.so.1
#1  0x00007ffff653c26d in ?? () from /home/user/binaryninja/libbinaryninjacore.so.1
#2  0x00007ffff63ea847 in ?? () from /home/user/binaryninja/libbinaryninjacore.so.1
#3  0x00007ffff62ccb02 in ?? () from /home/user/binaryninja/libbinaryninjacore.so.1
#4  0x00007ffff62ccdc1 in BNParseTypeString () from /home/user/binaryninja/libbinaryninjacore.so.1
#5  0x00000000005240ad in BinaryNinja::BinaryView::ParseTypeString(std::string const&, BinaryNinja::QualifiedNameAndType&, std::string&) ()
#6  0x0000000000517e2e in CreateTypeDialog::createType() ()
#7  0x00007ffff44e4cef in QMetaObject::activate(QObject*, int, int, void**) () from /home/user/binaryninja/libQt5Core.so.5
#8  0x00007ffff59874d2 in QAbstractButton::clicked(bool) () from /home/user/binaryninja/libQt5Widgets.so.5
#9  0x00007ffff57048c5 in ?? () from /home/user/binaryninja/libQt5Widgets.so.5
#10 0x00007ffff5705624 in ?? () from /home/user/binaryninja/libQt5Widgets.so.5
#11 0x00007ffff57057c4 in QAbstractButton::mouseReleaseEvent(QMouseEvent*) () from /home/user/binaryninja/libQt5Widgets.so.5
#12 0x00007ffff5650308 in QWidget::event(QEvent*) () from /home/user/binaryninja/libQt5Widgets.so.5
#13 0x00007ffff560e04c in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from /home/user/binaryninja/libQt5Widgets.so.5
#14 0x00007ffff5612f0e in QApplication::notify(QObject*, QEvent*) () from /home/user/binaryninja/libQt5Widgets.so.5
#15 0x00007ffff44bbe80 in QCoreApplication::notifyInternal2(QObject*, QEvent*) () from /home/user/binaryninja/libQt5Core.so.5
#16 0x00007ffff5611a33 in QApplicationPrivate::sendMouseEvent(QWidget*, QMouseEvent*, QWidget*, QWidget*, QWidget**, QPointer<QWidget>&, bool) () from /home/user/binaryninja/libQt5Widgets.so.5
#17 0x00007ffff566b5da in ?? () from /home/user/binaryninja/libQt5Widgets.so.5
#18 0x00007ffff566d6db in ?? () from /home/user/binaryninja/libQt5Widgets.so.5
#19 0x00007ffff560e04c in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from /home/user/binaryninja/libQt5Widgets.so.5
#20 0x00007ffff5612488 in QApplication::notify(QObject*, QEvent*) () from /home/user/binaryninja/libQt5Widgets.so.5
#21 0x00007ffff44bbe80 in QCoreApplication::notifyInternal2(QObject*, QEvent*) () from /home/user/binaryninja/libQt5Core.so.5
#22 0x00007ffff4dba97b in QGuiApplicationPrivate::processMouseEvent(QWindowSystemInterfacePrivate::MouseEvent*) () from /home/user/binaryninja/libQt5Gui.so.5
#23 0x00007ffff4dbc485 in QGuiApplicationPrivate::processWindowSystemEvent(QWindowSystemInterfacePrivate::WindowSystemEvent*) () from /home/user/binaryninja/libQt5Gui.so.5
#24 0x00007ffff4d9d6a3 in QWindowSystemInterface::sendWindowSystemEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /home/user/binaryninja/libQt5Gui.so.5
#25 0x00007fffe6fa4720 in ?? () from /home/user/binaryninja/qt/platforms/../../libQt5XcbQpa.so.5
#26 0x00007ffff25cc197 in g_main_context_dispatch () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#27 0x00007ffff25cc3f0 in ?? () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#28 0x00007ffff25cc49c in g_main_context_iteration () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#29 0x00007ffff450b4e7 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /home/user/binaryninja/libQt5Core.so.5
#30 0x00007ffff44b9efa in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () from /home/user/binaryninja/libQt5Core.so.5
#31 0x00007ffff57fc47c in QDialog::exec() () from /home/user/binaryninja/libQt5Widgets.so.5
#32 0x0000000000512b23 in TypeView::newType() ()
#33 0x000000000051421d in TypeView::keyPressEvent(QKeyEvent*) ()
#34 0x00007ffff5650544 in QWidget::event(QEvent*) () from /home/user/binaryninja/libQt5Widgets.so.5
#35 0x00007ffff574856e in QFrame::event(QEvent*) () from /home/user/binaryninja/libQt5Widgets.so.5
#36 0x00007ffff57ca053 in QAbstractScrollArea::event(QEvent*) () from /home/user/binaryninja/libQt5Widgets.so.5
#37 0x00007ffff560e04c in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from /home/user/binaryninja/libQt5Widgets.so.5
#38 0x00007ffff56122a9 in QApplication::notify(QObject*, QEvent*) () from /home/user/binaryninja/libQt5Widgets.so.5
#39 0x00007ffff44bbe80 in QCoreApplication::notifyInternal2(QObject*, QEvent*) () from /home/user/binaryninja/libQt5Core.so.5
#40 0x00007ffff566d6f3 in ?? () from /home/user/binaryninja/libQt5Widgets.so.5
#41 0x00007ffff560e04c in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from /home/user/binaryninja/libQt5Widgets.so.5
#42 0x00007ffff5612488 in QApplication::notify(QObject*, QEvent*) () from /home/user/binaryninja/libQt5Widgets.so.5
#43 0x00007ffff44bbe80 in QCoreApplication::notifyInternal2(QObject*, QEvent*) () from /home/user/binaryninja/libQt5Core.so.5
#44 0x00007ffff4db7738 in QGuiApplicationPrivate::processKeyEvent(QWindowSystemInterfacePrivate::KeyEvent*) () from /home/user/binaryninja/libQt5Gui.so.5
#45 0x00007ffff4dbc465 in QGuiApplicationPrivate::processWindowSystemEvent(QWindowSystemInterfacePrivate::WindowSystemEvent*) () from /home/user/binaryninja/libQt5Gui.so.5
#46 0x00007ffff4d9d6a3 in QWindowSystemInterface::sendWindowSystemEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /home/user/binaryninja/libQt5Gui.so.5
#47 0x00007fffe6fa4720 in ?? () from /home/user/binaryninja/qt/platforms/../../libQt5XcbQpa.so.5
#48 0x00007ffff25cc197 in g_main_context_dispatch () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#49 0x00007ffff25cc3f0 in ?? () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#50 0x00007ffff25cc49c in g_main_context_iteration () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#51 0x00007ffff450b4e7 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /home/user/binaryninja/libQt5Core.so.5
#52 0x00007ffff44b9efa in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () from /home/user/binaryninja/libQt5Core.so.5
#53 0x00007ffff44c1d9d in QCoreApplication::exec() () from /home/user/binaryninja/libQt5Core.so.5
#54 0x0000000000439244 in main ()
(gdb) i r
rax            0x7fffffffadc0	140737488334272
rbx            0x1	1
rcx            0x0	0
rdx            0x7ffff4032140	140737287233856
rsi            0x7ffff4032140	140737287233856
rdi            0x7fffe2caefd0	140736998338512
rbp            0x0	0x0
rsp            0x7fffffffadb0	0x7fffffffadb0
r8             0x7ffff4032158	140737287233880
r9             0x0	0
r10            0x7ffff353f5a0	140737275753888
r11            0x7ffff353eca0	140737275751584
r12            0x7ffff4032140	140737287233856
r13            0x7fffffffadc0	140737488334272
r14            0x7ffff6ed70a0	140737336144032
r15            0x7fffc77af540	140736540112192
rip            0x7ffff63dd7c3	0x7ffff63dd7c3
eflags         0x10206	[ PF IF RF ]
cs             0x33	51
ss             0x2b	43
ds             0x0	0
es             0x0	0
fs             0x0	0
gs             0x0	0
```
# debug + stderr flags
nothing interesting when I run as `./binaryninja --debug --stderr-log`

**Comment by psifertex** [comment]
Does this still happen? Are you adding the enum on a raw file, or an existing binary? I'm unable to reproduce it here.

**Comment by psifertex** [comment]
I'm pretty confident this is the same as #904 so I'm going to close this one though even though it's first since the other one describes the bug more accurately.

**Comment by bpotchik** [comment]
Fixed in build 1.1.1042-dev.
